nixos/modules/services/monitoring/librenms.nix

```diff
@@ -59,6 +54,16 @@ in
   options.services.librenms = with lib; {
     enable = mkEnableOption "LibreNMS network monitoring system";
 
+    package = mkOption {
+      type = types.package;
+      default = pkgs.librenms.override {
+        logDir = cfg.logDir;
+        dataDir = cfg.dataDir;
+      };
+      defaultText = "pkgs.librenms.override { logDir = cfg.logDir; dataDir = cfg.dataDir; }";
+      description = "LibreNMS package to use.";
+    };
```

Comment:
This can break things if the `logDir` and the `dataDir` in the override are different from the settings of the module. Therefore, I would prefer to do the overrides in the module like in these examples:

https://github.com/NixOS/nixpkgs/blob/7cd591618e0831316e1bed89a4924d88fd2efb1f/nixos/modules/services/networking/softether.nix#L8
https://github.com/NixOS/nixpkgs/blob/7cd591618e0831316e1bed89a4924d88fd2efb1f/nixos/modules/services/misc/guix/default.nix#L6

Reply:
I see the issue, the problem is I would love to have a way to expose the package that I don't have to manually configure to use `snmp-scan.py` and make sure it's in sync with the module:
https://github.com/rhoriguchi/nixos-setup/blob/09144a8ab216251cc6b60e9050e9c48acc8718dc/configuration/devices/headless/router/librenms.nix#L97-L99

How about adding some checks so that the package and the options have to be the same?

Reply:
Both of my examples still have a package option. They just do the package override in the module.

So I have no problem with the package option, I just want the package override to happen in the module.

Reply:
I've changed it to your proposal. It seems to still work since they get set here as default it will probably cause issues if you don't use default locations https://github.com/NixOS/nixpkgs/blob/de18c74d31da88b8c6919cd1046c67a11e11b2ac/pkgs/by-name/li/librenms/package.nix#L20

Reply:
The cron job I'm using is using the `librenms` package, but to find the config it needs to use the internal `dataDir` variable that is set on the package. If you now change the `services.librenms.dataDir` the package won't have the correct data dir set since the package only exposes the raw package without the overrides set in the module. In my use cases where I use the default, this will not cause any issues but if you change it, you might run into issues.

 https://github.com/NixOS/nixpkgs/blob/26abd85e2c1b78dc403cfa38aeb1efa31c5103f4/nixos/modules/services/monitoring/librenms.nix#L8-L11

```nix
services.cron.systemCronJobs = [
  "27 * * * * ${config.services.librenms.user} ${pkgs.python3}/bin/python ${config.services.librenms.package}/snmp-scan.py >> /dev/null 2>&1"
];
```

Reply:
Ah okay I understand. Than how about adding an read-only option that exposes the final package, like mediawiki does:
https://github.com/NixOS/nixpkgs/blob/9ad61cfc19c1f66e7d7d1ac9c71ae1b187065580/nixos/modules/services/web-apps/mediawiki.nix#L208

Reply:
Ahh nice did not know this is an option to make an option readonly. This perfectly solves this issue. Thank you for the review and inputs.

Reply:
Thank you for the contribution :)